examples/raw_upload/src/main.rs

```diff
@@ -21,3 +21,35 @@ fn index() -> &'static str {
 fn main() {
     rocket::ignite().mount("/", routes![index, upload]).launch();
 }
+
+#[cfg(test)]
+mod test {
+    use super::rocket;
+    use rocket::testing::MockRequest;
+    use rocket::http::{Status, ContentType};
+    use rocket::http::Method::Post;
+    use rocket::response::NamedFile;
+    use std::io::Read;
+
+    #[test]
+    fn test_raw_upload() {
+        let text_to_upload = "Text to be uploaded!".to_string();
+
+        // Upload the body
+        let rocket =
+            rocket::ignite().mount("/", routes![super::index, super::upload]);
+        let mut request = MockRequest::new(Post, "/upload")
+            .header(ContentType::Plain)
+            .body(&text_to_upload);
+        let response = request.dispatch_with(&rocket);
+        assert_eq!(response.status(), Status::Ok);
+
+        // Ensure we find the body in the /tmp/upload.txt file
+        let mut found_text = String::new();
+        let bytes_read = NamedFile::open("/tmp/upload.txt")
+            .and_then(|mut f| f.read_to_string(&mut found_text));
+
+        assert_eq!(bytes_read.expect("Unable to Read"), text_to_upload.capacity());
```

Comment:
Why are we checking against `capacity` and not `len`? Also, doesn't the next test for equality encompass this test? I think we can kill this line.